src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java

```diff
@@ -65,6 +65,9 @@ protected void onCreate(Bundle savedInstanceState) {
                     throw new IllegalArgumentException(
                             "You must provide the requested fragment type as an extra.");
             }
+        } else {
```

Comment:
I still don't see the point of this. The only way it could happen if someone forgets to set the intent action in new code, and in that case, we should throw an AssertionError.